src/com/ms/silverking/collection/CollectionUtil.java

```diff
@@ -2,121 +2,151 @@
-            sb.append(separator);
-        }
-        
-        sb.delete(sb.length()-separator.length(), sb.length());
-        sb.append(endBrace);   
+    return false;
+  }
+
+  public static <K, V> String mapToString(Map<K, V> map) {
+    return mapToString(map, defaultStartBrace, defaultEndBrace, defaultMapString, defaultMapEntrySeparator,
+        defaultEmptyDef);
+  }
+
+  public static <K, V> String mapToString(Map<K, V> map, String startBrace, String endBrace, String mapString,
+      String separator, String emptyDef) {
+    if (map.isEmpty())
+      return emptyDef;
 
-        return sb.toString();
+    StringBuilder sb = new StringBuilder();
+    sb.append(startBrace);
+
+    for (Map.Entry<K, V> entry : map.entrySet()) {
+      sb.append(entry.getKey() + mapString + entry.getValue());
+      sb.append(separator);
     }
- 
-    public static <K> Set<String> stringSet(Set<K> s) {
-        ImmutableSet.Builder<String>    ss;
-        
-        ss = ImmutableSet.builder();
-        for (K k : s) {
-            ss.add(k.toString());
-        }
-        return ss.build();
+
+    sb.delete(sb.length() - separator.length(), sb.length());
+    sb.append(endBrace);
+
+    return sb.toString();
+  }
+
+  public static <K> Set<String> stringSet(Set<K> s) {
+    ImmutableSet.Builder<String> ss;
+
+    ss = ImmutableSet.builder();
+    for (K k : s) {
+      ss.add(k.toString());
     }
-    
-    public static <K,V> SetMultimap<V,K> transposeSetMultimap(SetMultimap<K,V> m) {
-        SetMultimap<V,K>   tm;
-        
-        tm = HashMultimap.create();
-        for (K k : m.keySet()) {
-            for (V v : m.get(k)) {
-                tm.put(v, k);
-            }
-        }
-        return tm;
+    return ss.build();
+  }
+
+  public static <K, V> SetMultimap<V, K> transposeSetMultimap(SetMultimap<K, V> m) {
+    SetMultimap<V, K> tm;
+
+    tm = HashMultimap.create();
+    for (K k : m.keySet()) {
+      for (V v : m.get(k)) {
+        tm.put(v, k);
+      }
     }
-    
-    public static Set<String> parseSet(String def, String pattern) {
-        if (def != null) {
-            def = def.trim();
-            if (def.startsWith(defaultStartBrace)) {
-                def = def.substring(defaultStartBrace.length());
-                if (def.endsWith(defaultEndBrace)) {
-                    def = def.substring(0, def.length() - defaultEndBrace.length());
-                }
-            }
-            return ImmutableSet.copyOf(def.split(pattern));
-        } else {
-            return ImmutableSet.of();
+    return tm;
+  }
+
+  public static Set<String> parseSet(String def, String pattern) {
+    if (def != null) {
+      def = def.trim();
+      if (def.startsWith(defaultStartBrace)) {
+        def = def.substring(defaultStartBrace.length());
+        if (def.endsWith(defaultEndBrace)) {
+          def = def.substring(0, def.length() - defaultEndBrace.length());
         }
+      }
+      return ImmutableSet.copyOf(def.split(pattern));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;pattern&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;def&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, reduce the method&#39;s visibility.

Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).